prov/verbs/src/fi_verbs.c

```diff
@@ -970,7 +1006,10 @@ static int fi_ibv_getinfo(uint32_t version, const char *node, const char *servic
 		goto err2;
 
 	if (hints) {
-		ret = fi_ibv_check_hints(hints, verbs_info);
+		/* TODO if we do not bind to a specific device we have to check 
+		 * check_info list and return those fi_info structures that match hints
+		 */
```

Comment:
Long term, this is what I think is needed:

If the app does not specify a source address, a list should be returned.  At least 1 entry in that list should _not_ be associated with a device.  The src_addr for that entry should contain a wildcard network address.  Entries that are associated with a specific device should contain a src_addr with a specific network address, but wildcard transport.